oauth/src/main/java/ch/cyberduck/core/oauth/OAuth2AuthorizationService.java

```diff
@@ -165,11 +166,32 @@ private TokenResponse authorizeWithCode(final Host bookmark, final LoginCallback
             );
         }
         // Start OAuth2 flow within browser
+        final ClientParametersAuthentication clientParameters;
+        if(StringUtils.isBlank(clientid)) {
+            if(null != new HostPreferences(bookmark).getProperty("oauth.clientid")) {
+                clientParameters = new ClientParametersAuthentication(new HostPreferences(bookmark).getProperty("oauth.clientid"), StringUtils.EMPTY);
+            }
+            else {
+                final Credentials clientid = prompt.prompt(bookmark,
+                        LocaleFactory.localizedString("OAuth Client ID", "Credentials"),
+                        LocaleFactory.localizedString("Provide additional login credentials", "Credentials"),
+                        new LoginOptions(bookmark.getProtocol())
+                                .user(true).password(true)
+                                .passwordPlaceholder("OAuth Client ID"));
+                if(clientid.isSaved()) {
+                    bookmark.setProperty("oauth.clientid", clientid.getPassword());
+                }
+                clientParameters = new ClientParametersAuthentication(clientid.getUsername(), StringUtils.EMPTY);
```

Comment:
PKCE needs to be enabled in the OC OAuth profile.

Reply:
Use of PKCE is enabled by default. Connection profile draft can be found in iterate-ch/profiles#48